src/ipc/process.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_IPC_PROCESS_H
+#define BITCOIN_IPC_PROCESS_H
+
+#include <memory>
+#include <string>
+
+namespace ipc {
+class Protocol;
+
+//! IPC process interface for spawning bitcoin processes and serving requests
+//! in processes that have been spawned.
+//!
+//! There will different implementations of this interface depending on the
+//! platform (e.g. unix, windows).
+class Process
+{
+public:
+    virtual ~Process() = default;
+
+    //! Spawn process and return socket file descriptor for communicating with
+    //! it.
+    virtual int spawn(const std::string& new_exe_name, int& pid) = 0;
+
+    //! Wait for spawned process to exit and return its exit code.
+    virtual int wait(int pid) = 0;
+
+    //! Serve requests if current process is a spawned subprocess. Blocks until
+    //! socket for communicating with the parent process is disconnected.
+    virtual bool serve(int& exit_status) = 0;
```

Comment:
For each method of this interface it might be good to prefix them by which side of the process relationship it's expected to be used, `parent_spawn`, `parent_wait`, `spawn_serve`.

But I don't think it makes sense to split it further because we might have in the future set of process with grandchild relationship `bitcoin-gui` -> `bitcoin-wallet` -> `bitcoin-keyholder` ?

Reply:
New documentation is better, reading them I grasped those methods usage at first read!